src/sampler/single_particle_dists/maxwell_boltzmann.jl

```diff
@@ -0,0 +1,58 @@
+struct MaxwellBoltzmannParticle{D,P,T,DIST} <: SingleParticleDistribution
+    dir::D
+    part::P
+    temperature::T
+    rho_dist::DIST
+
+    function MaxwellBoltzmannParticle(
+        dir::D, particle::P, temperature::T
+    ) where {D<:ParticleDirection,P<:AbstractParticleType,T<:Real}
+        a = sqrt(mass(particle) * temperature)
+        return new{D,P,T,MaxwellBoltzmann{T}}(
+            dir, particle, temperature, MaxwellBoltzmann(a)
+        )
+    end
+end
+
+_particle(d::MaxwellBoltzmannParticle) = d.part
+_particle_direction(d::MaxwellBoltzmannParticle) = d.dir
+temperature(d::MaxwellBoltzmannParticle) = d.temperature
+
+function QEDevents._weight(
+    d::MaxwellBoltzmannParticle{D,P,T}, ps::ParticleStateful
+) where {D,P,T}
```

Comment:
It looks like `{D, P, T}` aren't actually needed. But maybe we should check that the `ParticleStateful`'s direction matches?

```suggestion
function QEDevents._weight(
    d::MaxwellBoltzmannParticle{D}, ps::ParticleStateful{D}
) where {D}
```

This would disallow using `UnknownDirection` as a sort of wildcard, which could probably be solved by some sort of `Union{D, UnknownDirection}`

Reply:
You are right, this is fixed in 41314f5. For `ParticleStateful` the direction and the particles are now part of the signature. 

However, why is `UnknownDirection` disallowed? Isn't it possible to have `ParticleStateful{UnkownDirection}`?

Reply:
`UnknownDirection` is allowed on its own. I meant it couldn't be used as a wildcard like this, like having a `MaxwellBoltzmannParticle{UnknownDirection}` with a `ParticleStateful{Incoming}` for example.

Reply:
I think there should be no *wildcard* here, but the direction particle-stateful should match the one from the distribution.